* major() and minor() need both sys/types.h and sys/mkdev.h included on Solaris

--- configure.ac
+++ configure.ac
@@ -167,7 +167,7 @@
 AC_CHECK_HEADERS([locale.h paths.h poll.h pwd.h regex.h stdarg.h])
 AC_CHECK_HEADERS([stdint.h stdlib.h string.h sys/acl.h sys/ioctl.h])
 AC_CHECK_HEADERS([sys/param.h sys/poll.h sys/select.h sys/time.h sys/utime.h])
-AC_CHECK_HEADERS([time.h unistd.h utime.h wchar.h wctype.h])
+AC_CHECK_HEADERS([time.h unistd.h utime.h wchar.h wctype.h sys/mkdev.h])
 
 # Checks for libraries.
 AC_ARG_WITH([zlib],
--- cpio/cpio.c
+++ cpio/cpio.c
@@ -60,6 +60,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 
 #include "cpio.h"
 #include "matching.h"
